mysql-test/mysql-test-run.pl

```diff
@@ -4128,15 +4128,15 @@ ($$)
       # ----------------------------------------------------
       # Check if it was an expected crash
       # ----------------------------------------------------
-      my $check_crash = check_expected_crash_and_restart($proc);
-      if ($check_crash == 0) # unexpected exit/crash of $proc
+      my $check_crash = check_expected_crash_and_restart($wait_for_proc);
+      if ($check_crash == 0) # unexpected exit/crash of $wait_for_proc
       {
         $expected_exit = 0;
```

Comment:
I'd use `goto SRVDIED` here without any `$expected_exit` flag (would've saved the time I spent on understanding how `$expected_exit` works). But it's a matter of taste, so up to you.

Anyway, this looks wrong. Old code (before your first patch) proceeded to SRVDIED here with `$proc` being the process that unexpectedly crashed. But here `$wait_for_proc` is the process that unexpectedly crashed and `$proc` is the process that exited last. You need `$proc=$wait_for_proc;` here.